include/igl/generate_sparse_marching_cubes_grid.cpp

```diff
@@ -0,0 +1,152 @@
+#include "generate_sparse_marching_cubes_grid.h"
+
+#include <unordered_map>
+#include <array>
+#include <vector>
+
+
+struct RowVector3iHash {
```

Comment:
If these are only used once in the function below then you could consider using lambdas. If you think they'll be useful else where then add a .h/.cpp for each function.